group-coordinator/src/test/java/org/apache/kafka/coordinator/group/AssignmentTestUtil.java

```diff
@@ -86,21 +86,19 @@ public static void assertAssignment(
     }
 
     /**
-     * Generate a reverse look up map of partition to member target assignments from the given member spec.
+     * Generate a reverse look up map of partition to member target assignments from the given metadata.
      *
-     * @param memberSpec        A map where the key is the member Id and the value is an
-     *                          AssignmentMemberSpec object containing the member's partition assignments.
+     * @param members       The member subscription specs.
      * @return Map of topic partition to member assignments.
      */
     public static Map<Uuid, Map<Integer, String>> invertedTargetAssignment(
-        Map<String, AssignmentMemberSpec> memberSpec
+        Map<String, MemberSubscriptionSpecImpl> members
     ) {
         Map<Uuid, Map<Integer, String>> invertedTargetAssignment = new HashMap<>();
-        for (Map.Entry<String, AssignmentMemberSpec> memberEntry : memberSpec.entrySet()) {
-            String memberId = memberEntry.getKey();
-            Map<Uuid, Set<Integer>> topicsAndPartitions = memberEntry.getValue().assignedPartitions();
+        for (String memberId : members.keySet()) {
```

Comment:
nit: It looks like that we could keep the previous code for the iteration. The only change should be replacing `assignedPartitions` by `memberAssignment`.